src/plugins/dashboard/public/application/hooks/use_dashboard_app_state.ts

```diff
@@ -148,6 +150,15 @@ export const useDashboardAppState = ({
       if (canceled || !loadSavedDashboardResult) return;
       const { savedDashboard, savedDashboardState } = loadSavedDashboardResult;
 
+      // If the saved dashboard is an alias match, then we will redirect
+      if (savedDashboard.outcome === 'aliasMatch') {
+        if (screenshotModeService?.isScreenshotMode()) {
+          scopedHistory().replace(savedDashboard.getFullPath());
```

Comment:
How can I test this? When I load the dashboard and click the "Full screen" button, it doesn't add anything to the URL. And obviously by the point the page has loaded for an aliasMatch, if we aren't already in screenshot mode, the other redirect has already happened.

Do I actually need to generate a report to exercise this?

Reply:
Screenshot mode is a standalone plugin.  There's a method on it to set whether you are in screenshot mode or not.  Reporting sets it when it's doing its thing so you can generate a report.  

Or, I think you can fake it by setting this key in localStorage `__KBN_SCREENSHOT_MODE_ENABLED_KEY__` to 'true'